src/main/java/task_modification_use_case/TaskModificationController.java

```diff
@@ -0,0 +1,16 @@
+package task_modification_use_case;
+
+public class TaskModificationController {
+    private TaskModificationInputBoundary taskModificationInputBoundary;
+
```

Comment:
Fixed the logic error in the interactor, which was searching for a task with the old key instead of the key that we wanted to change the title of a given task too. I have shortened the instance variable names where appropriate. View Model will be implemented soon. I am unsure of what other SOLID violations you are talking about, and would appreciate some clarification.

Reply:
The SOLID violation I was talking about was in TaskFactory.

Reply:
The methods in TaskFactory which violated SOLID principles have been removed.